app/src/components/utils.js

```diff
@@ -61,42 +61,44 @@ const utils = {
 
   /**
    * @function getBucket
-   * Acquire core S3 bucket credential information with graceful default fallback
-   * @param {string} [bucketId=undefined] An optional bucketId to lookup
-   * @param {boolean} [throwable=false] Throws an error if no `bucketId` is found
+   * Acquire core S3 bucket credential information from database or configuration
+   * @param {string} [bucketId=undefined] An optional bucket ID to query database for bucket
    * @returns {object} An object containing accessKeyId, bucket, endpoint, key,
    * region and secretAccessKey attributes
-   * @throws If there are no records found with `bucketId` and `throwable` is true
+   * @throws If there are no records found with `bucketId` or, if `bucketId` is undefined,
+   * no bucket details exist in the configuration
    */
-  async getBucket(bucketId = undefined, throwable = false) {
-    const data = {
-      accessKeyId: config.get('objectStorage.accessKeyId'),
-      bucket: config.get('objectStorage.bucket'),
-      endpoint: config.get('objectStorage.endpoint'),
-      key: config.get('objectStorage.key'),
-      region: DEFAULTREGION,
-      secretAccessKey: config.get('objectStorage.secretAccessKey')
-    };
-
-    if (bucketId) {
-      // Function scoped import to avoid circular dependencies
-      const { bucketService } = require('../services');
-
-      try {
+  async getBucket(bucketId = undefined) {
+    try {
+      const data = { region: DEFAULTREGION };
+      if (bucketId) {
+        // Function scoped import to avoid circular dependencies
+        const { bucketService } = require('../services');
```

Comment:
I'm wondering if we should change this to be a deeper scoped import...
```js
const { read } = require('../services/bucket');
const bucketData = await read(bucketId);
```

Reply:
seems like a good idea. tested and it works